fk/keycreate.go

```diff
@@ -248,11 +250,18 @@ func verifyEmailMatchesKeyInAPI(
 
 	retrievedKey, err := pgpkey.LoadFromArmoredPublicKey(armoredKey)
 	if err != nil {
-		return false, fmt.Errorf("Failed to load armored key: %v", err)
+		return false, fmt.Errorf("failed to load armored key: %v", err)
 	}
 	if retrievedKey.Fingerprint() == fingerprint {
 		return true, nil
 	}
+	retrievedEmail, err := retrievedKey.Email()
+	if err != nil {
+		return false, fmt.Errorf("couldn't get email for retrieved key: %s", err)
+	}
+	if retrievedEmail == email {
```

Comment:
Unfortunately this logic isn't correct: if the key has multiple email addresses, `Email()` chooses the most "primary" one. So it's perfectly valid for `retrievedEmail` not to match `